x-pack/plugins/canvas/canvas_plugin_src/renderers/markdown/index.js

```diff
@@ -7,9 +7,38 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import Markdown from 'markdown-it';
+import mila from 'markdown-it-link-attributes';
+import { ELASTIC_WEBSITE_URL } from '../../../public/lib/documentation_links';
+import { escapeRegExp } from '../../../common/lib/escape_reg_exp';
+
+const ELASTIC_LINK = new RegExp(
+  `^${escapeRegExp(ELASTIC_WEBSITE_URL).replace('https', 'https?')}?`
+);
+const NON_ELASTIC_EXTERNAL_LINK = new RegExp(
+  `(?!^${escapeRegExp(ELASTIC_WEBSITE_URL)})((http|https):\\/\\/[\\w\\.\\/\\-=?#]+)`
```

Comment:
Parsing URLs using regular expressions generally makes me feel uneasy, because regexs are so easy to get wrong, I'd feel more comfortable if we were using [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) or a polyfill to do this instead.

Do we intend to have `\\/` in the rexex twice? If we're looking for a `/` we should only have to escape it once.
Also, `[\\w\\.\\/\\-=?#]` appears to be double escaping most characters.

The plugin itself seems to be rather simplistic, doing this not using regular expression might be a reasonable option: https://github.com/crookedneighbor/markdown-it-link-attributes/blob/master/index.js

Reply:
++